Lib/test/test_zipfile/test_core.py

```diff
@@ -3447,6 +3447,53 @@ def test_too_short(self):
         self.assertEqual(
             b"zzz", zipfile._Extra.strip(b"zzz", (self.ZIP64_EXTRA,)))
 
+class StoredZipExtFileRandomAccessTest(unittest.TestCase):
+    def test_random_access(self):
+        from _pyio import BytesIO
+        class StatIO(BytesIO):
+            def __init__(self):
+                super().__init__()
+                self.bytes_read = 0
+
+            def read(self, size=-1):
+                bs = super().read(size)
+                self.bytes_read += len(bs)
+                return bs
+
+            def get_bytes_read(self):
+                return self.bytes_read
+
+        sio = StatIO()
+        # 100000 bytes
+        txt = b'0123456789'*10000
+
+        # Check seek on a file
+        with zipfile.ZipFile(sio, "w", compression=zipfile.ZIP_STORED) as zipf:
+            zipf.writestr("foo.txt", txt)
+
+        with zipfile.ZipFile(sio, "r") as zipf:
+            with zipf.open("foo.txt", "r") as fp:
+                br = sio.get_bytes_read()
+                fp.seek(50000, os.SEEK_CUR)
+                self.assertEqual(sio.get_bytes_read() - br, 0, 'seek produces redundant read!')
```

Comment:
```suggestion
                self.assertEqual(sio.bytes_read, oldcount, 'seek produces redundant read!')
```

That way we know exactly the values of each operand in case of a failure.